magento/magento2
# Fix issues with Customer Data when form contains element with name `method`

### Description (*)

The path to verify `<form method="">` value in `customer-data.js` (`Magento_Customer`) is glitchy and when there's an element with `name="method"` in the submitted form, the `event.target.method` refers to element, not attribute.

![image](https://github.com/magento/magento2/assets/1639941/3464b36e-629a-4dad-9723-c235f04c2711)




### Related Pull Requests


### Fixed Issues (if relevant)

Please create one.

### Manual testing scenarios (*)
1. Open any form (e.g. Customer Registration form),
2. Open Browser console and replace `name=""` of any element (e.g. "Remember me") to `name="method"`.
3. Set breakpoint at `customer-data.min.js` line after `$(document).on('submit', function(event) {`. 
4. Expect that `if ()` section (1 line below) uses `<form method="">`  - not just created element.

### Questions or comments

We noticed the issue while troubleshooting Paradoxlabs Firstdata extension issues.

![image](https://github.com/magento/magento2/assets/1639941/4aace37a-eaa3-4044-a9ca-60f3123edaf3)

CC @rhoerr 



### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#38484: Fix issues with Customer Data when form contains element with name `method`
